tools/release

```diff
@@ -0,0 +1,210 @@
+# TODO What should we use?  I pulled this from the deb package metadata
+EMAIL=hello@timescale.com
+GIT_AUTHOR_NAME=tools/release
+GIT_COMMITTER_NAME=$GIT_AUTHOR_NAME
+export EMAIL GIT_AUTHOR_NAME GIT_COMMITTER_NAME
+
+BRANCH_BASENAME=forge-stable-
+CONTROL=extension/timescaledb_toolkit.control
+TOML=extension/Cargo.toml
+
+. tools/dependencies.sh
+
+set -ex
+
+# TODO Install these into timescaledev/toolkit-builder image and delete this block.
+if [ "$1" = setup ]; then
+    # Install cargo set-version (and cargo install is not idempotent).
+    if ! cargo help set-version > /dev/null; then
+        cargo install --version =$CARGO_EDIT cargo-edit
+    fi
+    # Install gh
+    gh=`basename $GH_DEB_URL`
+    curl -LO $GH_DEB_URL
+    sha256sum -c - <<EOF
+$GH_DEB_SHA256  $gh
+EOF
+    sudo dpkg -i $gh
+    exit
+fi
+
+print() {
+    printf '%s\n' "$*"
+}
+
+die() {
+    st=${?:-0}
+    if [ $st -eq 0 ]; then
+        st=2
+    fi
+    print "$*" >&2
+    exit $st
+}
+
+usage() {
+    die 'release [-n] -version VERSION COMMIT'
+}
+
+# Return 0 iff working directory is clean.
+# Also prints any diff.
+assert_clean() {
+    $nop git diff --exit-code
+}
+
+# Return 0 iff working directory is dirty.
+# Also prints any diff.
+assert_dirty() {
+    [ -n "$nop" ] && return
+    ! assert_clean
+}
+
+# Use start_commit, commit, and finish_commit to safely build a commit from
+# multiple automated edits.
+# - start_commit [file names]
+#   Start a commit with the named changed files.
+#   Any other edited file (dirty directory after commit) is an error.
+# - commit [file names]
+#   Amend the commit after each automated edit.
+#   Any other edited file (dirty directory after commit) is an error.
+# - finish_commit MESSAGE
+#   Finalize the commit with the commit message MESSAGE.
+#   Any edited files is an error.
+start_commit() {
+    [ -z "$_PENDING_COMMIT" ] || die 'BUG: start_commit called twice'
+    _PENDING_COMMIT=1
+    $nop git add "$@"
+    $nop git commit -m pending
+    assert_clean || die "working directory should be clean after commit $@"
+}
+
+commit() {
+    [ -n "$_PENDING_COMMIT" ] || die 'BUG: commit called without start_commit'
+    $nop git add "$@"
+    $nop git commit --no-edit --amend
+    assert_clean || die "working directory should be clean after commit $@"
+}
+
+finish_commit() {
+    [ -n "$_PENDING_COMMIT" ] || die 'BUG: finish_commit called without start_commit'
+    assert_clean || die "working directory should be clean to finish commit '$1'"
+    _PENDING_COMMIT=
+    (export GIT_COMMITTER_DATE="`date`" && $nop git commit --no-edit --amend "--date=$GIT_COMMITTER_DATE" -m "$1")
+}
+
+# Super simple option processing.
+while [ $# -gt 0 ]; do
+    arg=$1
+    shift
+    case "$arg" in
+        -n)
+            dry_run_flag=--dry-run
+            nop=:
+            ;;
+
+        -y)
+            # Ignored - gives us something to pass in the Github Action where default is no-op.
```

Comment:
This is confusing and gross, but I just don't have a better idea!  It gets the job done...